src/confluent/tools/handlers/metrics/get-confluent-cloud-metrics-handler.ts

```diff
@@ -0,0 +1,139 @@
+import { z } from "zod";
+import { ToolName } from "@src/confluent/tools/tool-name.js";
+import { logger } from "@src/logger.js";
+import { BaseToolHandler } from "../../base-tools.js";
+import { ClientManager } from "@src/confluent/client-manager.js";
+
+const resourceIdSchema = z.object({
+  "resource.kafka.id": z
+    .array(z.string().regex(/^lkc-/))
+    .optional()
+    .default(
+      process.env.KAFKA_CLUSTER_ID ? [process.env.KAFKA_CLUSTER_ID] : [],
+    ),
+  "resource.schema_registry.id": z.array(z.string().regex(/^lsrc-/)).optional(),
+  "resource.ksql.id": z.array(z.string().regex(/^lksqlc-/)).optional(),
+  "resource.compute_pool.id": z.array(z.string().regex(/^lfcp-/)).optional(),
+  "resource.connector.id": z.array(z.string().regex(/^lcc-/)).optional(),
+});
+
+const getConfluentCloudMetricsArguments = z.object({
+  resourceIds: resourceIdSchema
+    .partial()
+    .default(
+      process.env.KAFKA_CLUSTER_ID
+        ? { "resource.kafka.id": [process.env.KAFKA_CLUSTER_ID] }
+        : {},
+    )
+    .refine((obj) => Object.keys(obj).length > 0, {
+      message: "At least one resource type must be specified in resourceIds.",
+    }),
+  intervalStart: z.string().optional(),
+  intervalEnd: z.string().optional(),
+});
```

Comment:
Follow the getEnsuredParam pattern?

Try to avoid using `process.env.*` directly as we lose typing